ChangeLog.d/fix-aesni-asm-clobbers.txt

```diff
@@ -0,0 +1,3 @@
+Bugfix
+   * Specify previously missed XMM register clobbers in AES-NI asm blocks.
```

Comment:
Not just XMM: for the one you'd spotted originally, it's general-purpose registers. But anyway that's more detail than matters in the changelog, and conversely we'd like the changelog to explain the impact. So something like [this](https://github.com/Mbed-TLS/mbedtls/blob/mbedtls-3.1.0/ChangeLog#L94C6-L94C30):

> Fix missing constraints on the AESNI inline assembly which is used on GCC-like compilers when building AES for generic x86_64 targets. This may have resulted in incorrect code with some compilers, depending on optimizations. Fixes #9819.

Note that the other reviewers (we need two reviewers for each pull request) may request more changes.

I can handle the changelog updates (and any other rework) if you'd like.

Reply:
Oh, sure, I didn't mean to include XMM in that changelog entry, but ended up copy-pasting with the commit message too much. And it should be a separate commit then.

When I fix this, OK to amend/force-push the previous commit, or should I strictly be adding commits?

Reply:
Amending the last commit would be fine here.

Reply:
Done. I used your suggested wording almost as-is, although I'm not sure about the word "generic": not all x86_64 CPUs support AES-NI, but maybe in Mbed TLS context this word is somehow appropriate.

Reply:
I meant “generic” as in compiling without a target option like `-maes`. I think that's the usual way to phrase this, though I do realize it's ambiguous here. If you build the library in its default configuration for a generic x86_64 target, the AESNI assembly gets built, and then it may or may not be executed at runtime depending on whether AESNI is present.